### DISCUSSION TITLE
What is the location of the GitHub Copilot source code repository?

### DISCUSSION THREAD

**Discussion by andy5995** [community/community#41055] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

General

### Body

Anyone have the link to the GitHub Copilot source code repo? I couldn't find it on the Copilot page.

**Comment by DSDmark** [comment, marked as answer]
Hi @andy5995,

Because it's not open source. I guess, That reason you can't find it's repo.

**Comment by jfullstackdev** [comment]
seems there is none at the official GitHub Org of GitHub

![image](https://user-images.githubusercontent.com/47092464/206381979-33568c16-54b6-4d4e-bb79-d5917692f613.png)

you can post your request here anyway

**Comment by andy5995** [comment]
Whoa! I'm shocked and disappointed at the same time. I always felt the whole point of GitHub was open source software.

>you can post your request here anyway

But @jdevstatic yes, I'd like to request that GitHub Copilot be open source. I think opening the source sends a clear signal that Microsoft and GitHub wholly support the concept of open source software. Though I'm not against Microsoft offering proprietary or closed software if that's their choice (because I don't believe software *should* be free; rather, I believe people should be free to choose how to license their software), but I strongly feel that most anything associated with GitHub such as Copilot *should* be open.  Maybe I'm biased, but I love open source software!

**Reply by DSDmark** [reply]
In 2018 Microsoft acquired GitHub, from that time, it was playing with **GitHub** from time to time like the case of **youtube-dl**. You know what I mean... :shushing_face:

**Comment by jfullstackdev** [comment]
not always, some programs, even though GitHub supports open source, will not always be open source

in my development company, like the other vloggers in tech too, will say, some can be open source but not all particularly if it contains tight security
